ejb3/src/main/java/org/jboss/as/ejb3/iiop/EjbCorbaServant.java

```diff
@@ -314,21 +324,59 @@ public OutputStream _invoke(final String opName, final InputStream in, final Res
                                 retVal = false;
                             }
                         } else {
-
-                            if (sc != null) {
-                                setSecurityContextOnAssociation(sc);
+                            if (this.securityDomain != null) {
+                                // an elytron security domain is available: authenticate and authorize the client before
+                                // invoking the component.
+                                ServerAuthenticationContext context = this.securityDomain.createNewAuthenticationContext();
+                                if (credential == sasCurrent) {
+                                    // TODO change this to use the SASCurrent as the credential when we have a realm that can act on it.
+                                    credential = principal != null ? principal.getName().toCharArray() : null;
+                                }
+                                PasswordGuessEvidence evidence = new PasswordGuessEvidence(credential != null ? (char[]) credential : null);
+                                try {
+                                    context.setAuthenticationName(principal != null ? principal.getName() : "");
+                                    if (context.verifyEvidence(evidence)) {
```

Comment:
If there is no evidence, this step can be skipped. IIRC authorization will fail if the authentication requires evidence.